Automattic/jetpack
# VideoPress: generate video image when selecting poster from frame

This PR triggers the process that generates the video poster image according to a timestamp value.
It's handled at the `useSyncData()` hook level. It checks and detects changes in the block attributes once the post saves.

Please keep in mind that image generation is an asynchronous process. The way to know when the image is ready to use will be addressed in follow-up tasks.

Fixes #

## Proposed changes:

* VideoPress: generate video image when selecting a poster from the frame

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable beta extensions
* Go to the block editor
* Create/edit a VideoPress video block instance
* Open the block sidebar
* Toggle the video frame poster section

<img width="324" alt="Screen Shot 2023-03-27 at 20 22 12" src="https://user-images.githubusercontent.com/77539/228089490-931ec70c-bdbd-4a6f-93a1-8c5ef232ca73.png">

* Select a frame by using the proper control
* Save the post
* Confirm the app detects the need to generate the image by looking at the debug() lines

<img width="690" alt="Screen Shot 2023-03-27 at 20 27 19" src="https://user-images.githubusercontent.com/77539/228089489-23ca0f43-d0aa-4591-90c0-b68e89f823aa.png">

* Confirm the client performs a request to the `/wpcom/v2/videopress/<guide>/poster` endpoint to trigger the process that generates the video image:

request | response
----|----
<img width="771" alt="Screen Shot 2023-03-27 at 20 28 14" src="https://user-images.githubusercontent.com/77539/228089484-f92be649-bc3b-4376-8318-78f5ab57c3c1.png"> | <img width="888" alt="Screen Shot 2023-03-27 at 20 28 34" src="https://user-images.githubusercontent.com/77539/228089468-31b83b5c-22ee-43f2-8820-1215c2e253fe.png">
